File: lang/files/f.scala
import scala.util.matching.Regex
import collection.mutable.HashMap
import collection.immutable.TreeMap


val fname = "foo.txt"
//val lines = scala.io.Source.fromFile(fname).mkString
val lines = scala.io.Source.fromFile(fname).getLines

val h = new HashMap[String, Int]

val pattern = new Regex("""(.*)\s+(\d+)""", "str", "num")

for ( line  <- lines )  {
  val res = pattern.findFirstMatchIn(line).get
  
//  println(res.group("str") +", " + res.group("num"))
  val k = res.group("str")
  val v =  res.group("num").toInt
  if ( h.contains(k) ) {
    h(k) += v
  }
  else {
    h(k) = v
  }
}


val h_sorted = TreeMap(h.toSeq:_*)
for ( (k,v) <- h_sorted ) 
  println( k + "\t" + v)


/////////////////////////////////
/*
val pattern = new Regex("""(\w*) (\w*)""", "firstName", "lastName");
      val result = pattern.findFirstMatchIn("James Bond").get;
      println(result.group("lastName") + ", " + result.group("firstName"));
*/

File: lang/files/f.py
import re
import pprint
from collections import defaultdict

h = defaultdict()
p = re.compile(r'(\w+)\s+(.*)')


with open("foo.txt") as fh:
  for line in fh:
    m = p.match(line)
    if m:
      k = m.group(1)
      v = m.group(2)
      if k in h:
        h[k] += int(v)
      else:  
        h[k] = int(v)

#pprint.pprint( h )
#print h.items()

for k, v in h.items():
  print "%s\t%d" % (k, v)
